### PR TITLE
Expands on GatewayClass.spec.controller

### PR DESCRIPTION
**Pull request by bowei**
Adds more details on how to interpret the controller field, give
an example around how versioning will work.

Fixes: https://github.com/kubernetes-sigs/service-apis/issues/229

### CODE REVIEW

File: docs-src/concepts.md [hunk 1]
```diff
@@ -343,6 +343,29 @@ status:
     Message: "foobar" is an FooBar.
 ```
 
+### GatewayClass controller selection
+
+The `GatewayClass.spec.controller` field is used to determine whether
+or not a given `GatewayClass` is managed by the controller. The
+specifics of GatewayClass selection will depend on the controller(s)
```
Comment (jpeach) [inline docs-src/concepts.md:350]: Suggest using stronger language here:

> This format of this field is opaque and specific to a particular controller. Which GatewayClass is selected by a given `controller` field depends on how the various controller(s) in the cluster interpret this field.

**Comment by robscott** [comment]
@bowei This mostly LGTM, but agree with James that it would be worth adding slightly stronger language here.

**Comment by hbagdi** [comment]
@bowei Could you update the language here so that we can get this PR in?

**Comment by hbagdi** [comment]
Opened up #327 to move this forward.
/close

**Pull request closed without merging**